File: game.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>
#include <chrono>
#include <thread>

const int MAX_BULLETS = 6;
const int MAX_ENEMIES = 3;

class Bullet {
 public:
  Bullet() : damage_(10) {}

  int getDamage() const { return damage_; }

 private:
  int damage_;
};

class Gun {
 public:
  Gun(int maxBullets) : bullets_(maxBullets), maxBullets_(maxBullets) {}

  int getNumBullets() const { return bullets_; }
  void shoot() { bullets_--; }
  void reload() {
    std::cout << "Reloading..." << std::endl;
    std::this_thread::sleep_for(std::chrono::seconds(1));
    bullets_ = maxBullets_;
    std::cout << "Your gun has been reloaded!" << std::endl;
  }

 private:
  int bullets_;
  int maxBullets_;
};

class Player {
 public:
  Player() : health_(100), gun_(Gun(MAX_BULLETS)) {}

  int getHealth() const { return health_; }
  Gun& getGun() { return gun_; }

  void takeDamage(int damage) { health_ -= damage; }

  bool isAlive() const { return health_ > 0; }
  bool hasBullets() const { return gun_.getNumBullets() > 0; }

 private:
  int health_;
  Gun gun_;
};

int main() {
  srand(time(NULL));

  Player player;
  int enemies = MAX_ENEMIES;
  bool gameOver = false;

  while (player.isAlive() && enemies > 0 && !gameOver) {
    std::cout << "Bullets: " << player.getGun().getNumBullets() << std::endl;
    std::cout << "Enemies: " << enemies << std::endl;
    std::cout << "Health: " << player.getHealth() << std::endl;

    std::cout << "Enter a command (shoot, reload, exit): ";
    std::string command;
    std::cin >> command;

    if (command == "shoot") {
      if (player.hasBullets()) {
        player.getGun().shoot();

        if (rand() % 2 == 0) {
          enemies--;
          std::cout << "You hit an enemy!" << std::endl;
        } else {
          std::cout << "You missed!" << std::endl;
        }
      } else {
        std::cout << "You don't have any bullets left!" << std::endl;
      }
    } else if (command == "reload") {
      if (!player.hasBullets()) {
                player.getGun().reload();
      } else {
        std::cout << "Your gun is already full!" << std::endl;
      }
    } else if (command == "exit") {
      std::cout << "Thanks for playing, Made with chatGPT on 12/23/2022" << std::endl;
      gameOver = true;
      break;
    } else {
      std::cout << "Invalid command" << std::endl;
    }

    if (enemies > 0) {
      int damage = rand() % 10 + 10;
      std::cout << "An enemy attack you for " << damage << " damage!" << std::endl;
      player.takeDamage(damage);
    }
  }

  if (!player.isAlive()) {
    std::cout << "You have been killed by the enemies!" << std::endl;
  } else if (!gameOver) {
    std::cout << "You have defeated all of the enemies!" << std::endl;
  }

  return 0;
}

File: game.py
import random
import time

MAX_BULLETS = 6
MAX_ENEMIES = 3

class Bullet:
    def __init__(self):
        self.damage = 10

    def get_damage(self):
        return self.damage

class Gun:
    def __init__(self, max_bullets):
        self.bullets = max_bullets
        self.max_bullets = max_bullets

    def get_num_bullets(self):
        return self.bullets

    def shoot(self):
        self.bullets -= 1

    def reload(self):
        print("Reloading...")
        time.sleep(1)
        self.bullets = self.max_bullets
        print("Your gun has been reloaded!")

class Player:
    def __init__(self):
        self.health = 100
        self.gun = Gun(MAX_BULLETS)

    def get_health(self):
        return self.health

    def get_gun(self):
        return self.gun

    def take_damage(self, damage):
        self.health -= damage

    def is_alive(self):
        return self.health > 0

    def has_bullets(self):
        return self.gun.get_num_bullets() > 0

def main():
    random.seed()

    player = Player()
    enemies = MAX_ENEMIES
    game_over = False

    while player.is_alive() and enemies > 0 and not game_over:
        print(f"Bullets: {player.get_gun().get_num_bullets()}")
        print(f"Enemies: {enemies}")
        print(f"Health: {player.get_health()}")

        command = input("Enter a command (shoot, reload, exit): ")

        if command == "shoot":
            if player.has_bullets():
                player.get_gun().shoot()

                if random.randint(0, 1) == 0:
                    enemies -= 1
                    print("You hit an enemy!")
                else:
                    print("You missed!")
            else:
                print("You don't have any bullets left!")
        elif command == "reload":
            if not player.has_bullets():
                player.get_gun().reload()
            else:
                print("Your gun is already full!")
        elif command == "exit":
            print("Thanks for playing, Made with chatGPT on 12/23/2022")
            game_over = True
            break
        else:
            print("Invalid command")

        if enemies > 0:
            damage = random.randint(10, 19)
            print(f"An enemy attack you for {damage} damage!")
            player.take_damage(damage)

    if not player.is_alive():
        print("You have been killed by the enemies!")
    elif not game_over:
        print("You have defeated all of the enemies!")

if __name__ == "__main__":
    main()
